scrapy/spidermiddlewares/offsite.py

```diff
@@ -52,9 +53,18 @@ def get_host_regex(self, spider):
         allowed_domains = getattr(spider, 'allowed_domains', None)
         if not allowed_domains:
             return re.compile('') # allow all by default
+        for domain in allowed_domains:
+            url_pattern = re.compile("^https?://.*$")
+            if url_pattern.match(domain):
+                warnings.warn("allowed_domains accepts only domains, not URLs. Ignoring URL entry %s in allowed_domains." % domain, URLWarning)
```

Comment:
What about going the other way around, checking domain is a valid hostname string so it discards (and warns) on anything not matching.

Reply:
@dangra In this pull request, I wanted to keep it as simple as possible...

There was some discussion in the issue thread about whether removing invalid domain names should be done in addition to issuing a warning but the consensus seemed to have been that a new issue should be opened for that.

The issue I was trying to solve here was specifically about URLs vs. domain names, which was easy to do... But to implement your request to check for a valid domain name, we'd first need to have a discussion about how that regex should look like, how strict we should be...

I considered removing elements that matched the `url_pattern` regex from allowed_domains` but then it turned out that they are not being crawled anyway... and again, for actually removing instead of issuing a warning, there was the suggestion to open a separate issue.

Reply:
Fair point. thanks